plugins/misc/w_helicity.h

```diff
@@ -0,0 +1,186 @@
+                         Number aB_pt, Number aB_eta, Number aB_phi, Number aB_m)
+
+{
+  using namespace Framework;
+  static std::vector<std::pair<std::string, Number>> m_w_hel;
+  static int initialize = 0;
+  if (initialize == 0) {
+    m_w_hel.reserve(4); // exact size
+
+
+    // m_w_hel.emplace_back("f1A", -9999.);
+    // m_w_hel.emplace_back("f2A", -9999.);
+    m_w_hel.emplace_back("f1", -9999.);
+    m_w_hel.emplace_back("f2", -9999.);
+    // m_w_hel.emplace_back("f1paper", -9999.);
+    // m_w_hel.emplace_back("f2paper", -9999.);
+
+    ++initialize;
+  }
+
+  static std::array<Number, 32> arg;
+  if (arg[0]  == pTop_pt and arg[1]  == pTop_eta and arg[2]  == pTop_phi and arg[3]  == pTop_m and
+      arg[4]  == aTop_pt and arg[5]  == aTop_eta and arg[6]  == aTop_phi and arg[7]  == aTop_m and
+      arg[8]  == pLep_pt and arg[9]  == pLep_eta and arg[10] == pLep_phi and arg[11] == pLep_m and
+      arg[12] == aLep_pt and arg[13] == aLep_eta and arg[14] == aLep_phi and arg[15] == aLep_m and
+      arg[16] == pW_pt and arg[17] == pW_eta and arg[18] == pW_phi and arg[19] == pW_m and
+      arg[20] == aW_pt and arg[21] == aW_eta and arg[22] == aW_phi and arg[23] == aW_m and
+      arg[24] == pB_pt and arg[25] == pB_eta and arg[26] == pB_phi and arg[27] == pB_m and
+      arg[28] == aB_pt and arg[29] == aB_eta and arg[30] == aB_phi and arg[31] == aB_m)
+    return m_w_hel;
+
+  arg[0]  = pTop_pt; arg[1]  = pTop_eta; arg[2]  = pTop_phi; arg[3]  = pTop_m;
+  arg[4]  = aTop_pt; arg[5]  = aTop_eta; arg[6]  = aTop_phi; arg[7]  = aTop_m;
+  arg[8]  = pLep_pt; arg[9]  = pLep_eta; arg[10] = pLep_phi; arg[11] = pLep_m;
+  arg[12] = aLep_pt; arg[13] = aLep_eta; arg[14] = aLep_phi; arg[15] = aLep_m;
+  arg[16] = pW_pt; arg[17] = pW_eta; arg[18] = pW_phi; arg[19] = pW_m;
+  arg[20] = aW_pt; arg[21] = aW_eta; arg[22] = aW_phi; arg[23] = aW_m;
+  arg[24] = pB_pt; arg[25] = pB_eta; arg[26] = pB_phi; arg[27] = pB_m;
+  arg[28] = aB_pt; arg[29] = aB_eta; arg[30] = aB_phi; arg[31] = aB_m;
+
+
+  static TLorentzVector p4lab_pTop, p4lab_aTop, p4lab_pLep, p4lab_aLep, p4lab_pW, p4lab_aW,p4lab_pB, p4lab_aB;
+  p4lab_pTop.SetPtEtaPhiM(arg[0],  arg[1],  arg[2],  arg[3]);
+  p4lab_aTop.SetPtEtaPhiM(arg[4],  arg[5],  arg[6],  arg[7]);
+  p4lab_pLep.SetPtEtaPhiM(arg[8],  arg[9],  arg[10], arg[11]);
+  p4lab_aLep.SetPtEtaPhiM(arg[12], arg[13], arg[14], arg[15]);
+  p4lab_pW.SetPtEtaPhiM(arg[16], arg[17], arg[18], arg[19]);
+  p4lab_aW.SetPtEtaPhiM(arg[20], arg[21], arg[22], arg[23]);
+  p4lab_pB.SetPtEtaPhiM(arg[24], arg[25], arg[26], arg[27]);
+  p4lab_aB.SetPtEtaPhiM(arg[28], arg[29], arg[30], arg[31]);
+
+  // the necessary boosting such that lepton ~ top spin vector
+  const TLorentzVector p4lab_TT( p4lab_pTop + p4lab_aTop ); //calculates the t-tbar system 4-momenta
+  // const TLorentzVector p4lab_eb( p4lab_aLep + p4lab_pB ); //calculates the t-tbar system 4-momenta
+
+
+  const auto f_zmf_tt = [&p4lab_TT] (TLorentzVector p4) {
```

Comment:
identical to the version in spin plugin, can probably be promoted to a free function used by both

Reply:
probably best done in numeric_vector

Reply:
I was referring to the boosting function, which can take 2 arg instead of using p4_ttbar as a capture. But yes since it's removed, not needed anymore.